crypto/cms/cms_rsa.c

```diff
@@ -157,8 +157,10 @@ static int rsa_cms_encrypt(CMS_RecipientInfo *ri)
         }
         oaep->pSourceFunc = ossl_X509_ALGOR_from_nid(NID_pSpecified,
                                                      V_ASN1_OCTET_STRING, los);
-        if (oaep->pSourceFunc == NULL)
+        if (oaep->pSourceFunc == NULL) {
```

Comment:
I would change the scope of the variable los and then just free it  at the end if it is not NULL..
At the point where pSourceFunc is not NULL, set  los = NULL since it is now owned by oaep..

Reply:
botovq also suggested changing the scope.

This makes sense since then `los` is not a special case (all the other variables have function scope).

I will make that change.